src/js/pages/cart.ts

```diff
@@ -33,12 +34,32 @@ export default () => {
     });
   });
 
-  const cartContainer = document.querySelector<HTMLElement>(Theme.selectors.cart.container);
-
   if (cartContainer) {
     cartContainer.addEventListener('click', (event: Event) => {
       const eventTarget = event.target as HTMLElement;
 
+      const targetItem = eventTarget.closest('.cart__item');
+      const targetValue = targetItem?.querySelector('.js-cart-line-product-quantity') as HTMLInputElement | null;
+      const removeButton = targetItem?.querySelector('.remove-from-cart') as HTMLElement | null;
+
+      if (targetValue) {
+        if (eventTarget.classList.contains('js-increment-button')) {
+          if (targetValue.dataset.mode === 'confirmation' && Number(targetValue.value) < 1) {
+            if (removeButton) {
+              removeButton.click();
+            }
+          }
+        }
+
+        if (eventTarget.classList.contains('js-decrement-button')) {
+          if (targetValue.getAttribute('value') === '1' && targetValue.getAttribute('min') === '1') {
+            if (removeButton) {
+              removeButton.click();
+            }
+          }
+        }
+      }
```

Comment:
```suggestion
      if (!removeButton || !targetValue) {
        return;
      }
      
      if (eventTarget.classList.contains('js-increment-button')) {
        if (targetValue.dataset.mode === 'confirmation' && Number(targetValue.value) < 1) {
            removeButton.click();
        }
      }

      if (eventTarget.classList.contains('js-decrement-button')) {
        if (targetValue.getAttribute('value') === '1' && targetValue.getAttribute('min') === '1') {
            removeButton.click();
        }
      }
```

The conde can be simplified, the firs `if` avoid `3 ifs`